rcl_action/src/rcl_action/action_server_impl.h

```diff
@@ -33,7 +33,7 @@ typedef struct rcl_action_server_impl_t
   rcl_action_goal_handle_t ** goal_handles;
   size_t num_goal_handles;
   // Clock
-  rcl_clock_t clock;
+  rcl_clock_t * clock;
```

Comment:
If action_server doesn't own a clock, then it would require the user to keep it around with the action_server. Perhaps, there just needs to be a new function to properly copy a clock?

Reply:
> If action_server doesn't own a clock, then it would require the user to keep it around with the action_server.

I'm not sure the action server owns the clock. It looks like it doesn't call `rcl_clock_fini()` on the copy it made.

https://github.com/ros2/rcl/blob/fb190af758b7fae42050ddece6fa04cef91eda8f/rcl_action/src/rcl_action/action_server.c#L207-L255

Reply:
Oh fair point